core/src/main/java/org/apache/lucene/search/vectorhighlight/CustomFieldQuery.java

```diff
@@ -56,6 +58,11 @@ public CustomFieldQuery(Query query, IndexReader reader, boolean phraseHighlight
 
     @Override
     void flatten(Query sourceQuery, IndexReader reader, Collection<Query> flatQueries, float boost) throws IOException {
+        while (sourceQuery instanceof BoostQuery) {
```

Comment:
When `BoostQuery` need to parse to other query firstly, because `BoostQuery` also  can have `BoostingQuery`